File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/neighbourhood/Neighborhood.scala
package pl.edu.agh.multiscalemodelling.engine.logic.neighbourhood

import java.util

import pl.edu.agh.multiscalemodelling.engine.logic.{Cell, Point}
import pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition.BoundaryCondition
import pl.edu.agh.multiscalemodelling.processsimulation.naiveseedsgrowth.NaiveSeedsGrowthCell

trait Neighborhood {
  def findNeighbors(cells: util.List[NaiveSeedsGrowthCell], currentCell: Cell, boundaryCondition: BoundaryCondition, size: Point): util.List[Cell]
}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/neighbourhood/Neighborhood.java
package pl.edu.agh.multiscalemodelling.engine.logic.neighbourhood;

import pl.edu.agh.multiscalemodelling.engine.logic.Cell;
import pl.edu.agh.multiscalemodelling.engine.logic.Point;
import pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition.BoundaryCondition;

import java.util.List;

public abstract class Neighborhood {

    public abstract List<Cell> findNeighbors(List<Cell> cells, Cell currentCell, BoundaryCondition boundaryCondition, Point size);

}
